src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -1922,10 +1970,42 @@ protected function check_is_comment_content_allowed( $prepared_comment ) {
 			return true;
 		}
 
+		// Allow empty notes only when resolution metadata is valid.
+		if (
+			isset( $check['comment_type'] ) &&
+			'note' === $check['comment_type'] &&
+			isset( $check['meta']['_wp_note_status'] ) &&
+			in_array( $check['meta']['_wp_note_status'], array( 'resolved', 'reopen' ), true )
+		) {
+			return true;
+		}
+
 		/*
 		 * Do not allow a comment to be created with missing or empty
 		 * comment_content. See wp_handle_comment_submission().
 		 */
 		return '' !== $check['comment_content'];
 	}
+
+	/**
+	 * Check if post type supports notes.
+	 *
+	 * @param string $post_type Post type name.
+	 * @return bool True if post type supports notes, false otherwise.
+	 */
+	private function check_post_type_supports_notes( $post_type ) {
+		$supports = get_all_post_type_supports( $post_type );
+		if ( ! isset( $supports['editor'] ) ) {
+			return false;
+		}
+		if ( ! is_array( $supports['editor'] ) ) {
+			return false;
+		}
+		foreach ( $supports['editor'] as $item ) {
+			if ( is_array( $item ) && isset( $item['notes'] ) && true === $item['notes'] ) {
```

Comment:
```suggestion
			if ( ! empty( $item['notes'] ) ) {
```

We don't really need all these extra checks, https://3v4l.org/1H8DU